tests/ifs/ifs.c

```diff
@@ -156,149 +94,154 @@ static bool load_test_file(int dfd, int batch_fd, struct test *test, ifs_test_t
     return false;
 }
 
-static int scan_init(struct test *test)
+static int scan_common_init(struct test *test)
 {
-        /* allocate info struct */
-        ifs_test_t *ifs_info = malloc(sizeof(ifs_test_t));
-
-        /* see if driver is loaded, otherwise try to load it */
-        int ifs0 = open(PATH_SYS_IFS_BASE "intel_ifs_0", O_DIRECTORY | O_PATH | O_CLOEXEC);
-        if (ifs0 < 0) {
-                /* modprobe kernel driver, ignore errors entirely here */
-                pid_t pid = fork();
-                if (pid == 0) {
-                        execl("/sbin/modprobe", "/sbin/modprobe", "-q", "intel_ifs", NULL);
-
-                        /* don't print an error if /sbin/modprobe wasn't found, but
-                           log_debug() is fine (since the parent is waiting, we can
-                           write to the FILE* because it's unbuffered) */
-                        log_debug("Failed to run modprobe: %s", strerror(errno));
-                        _exit(errno);
-                } else if (pid > 0) {
-                        /* wait for child */
-                        int status, ret;
-                        do {
-                            ret = waitpid(pid, &status, 0);
-                        } while (ret < 0 && errno == EINTR);
-                } else {
-                        /* ignore failure to fork() -- extremely unlikely */
-                }
+        /* Get info struct */
+        ifs_test_t *ifs_info = test->data;
 
-                /* try opening again now that we've potentially modprobe'd */
-                ifs0 = open(PATH_SYS_IFS_BASE "intel_ifs_0", O_DIRECTORY | O_PATH | O_CLOEXEC);
-        }
+        /* see if driver is loaded */
+        char sys_path[PATH_LEN];
+        int n = snprintf(sys_path, PATH_LEN, PATH_SYS_IFS_BASE "%s", ifs_info->sys_dir);
+        assert(n < sizeof(sys_path));
+        int ifs_fd = kernel_driver_is_loaded(sys_path);
```

Comment:
Need to check here that it didn't return -1, otherwise the openat() below will fail and we'll get a cryptic "Could not open .../run_test for writing (not running as root?): Bad file descriptor".

So, please insert here:
```c
if (ifs_fd < 0) {
        saved_errno = errno;
        log_info("could not find IFS control files in %s: either IFS is not supported on this system"
                 " or this kernel does not support IFS (%m)", ifs_info->sys_dir);
        return -saved_errno;
}
```

I might add the modprobe error message later.

Reply:
Fixed.